### PR TITLE
"MGDSTRM-682:Fix alerts - (ScrapeProblem) there should be no critical…

### PR DESCRIPTION
**Pull request by ramu11**
… alerts firing when one AZ is down"

### CODE REVIEW

**Comment by ramu11** [comment]
What
Current Prometheus rule to scrap metric is very simple
Why
During multi-az fault-injection testing there were  critical alerts firing ScrapeProblem. During such testing no critical alerts should be firing. Either tweak the alerts so that they don't fire in this case, or lower the severity of them
How
Improved the Rule as per https://github.com/openshift/cluster-monitoring-operator/blob/10453dfc18efff118c4f171208fb88aa73df4f4c/assets/prometheus-k8s/rules.yaml#L2212-L2220

**Comment by ramu11** [comment]
@david-martin  please review

File: install/resources/monitoring-cluster/prometheus/prometheus-rules-groups.yaml [hunk 1]
```diff
@@ -306,13 +306,13 @@ groups:
       summary: 'Kafka offline log directories'
       description: 'There are {{ $value }} offline log directories on {{ $labels.kubernetes_pod_name }}'
   - alert: ScrapeProblem
-    expr: up{kubernetes_namespace!~"openshift-.+",kubernetes_pod_name=~".+-kafka-[0-9]+"} == 0
+    expr: 100 * (count(up{kubernetes_namespace!~"openshift-.+",kubernetes_pod_name=~".+-kafka-[0-9]+"} == 0) / count(up{kubernetes_namespace!~"openshift-.+",kubernetes_pod_name=~".+-kafka-[0-9]+"})) > 3
```
Comment (david-martin) [inline install/resources/monitoring-cluster/prometheus/prometheus-rules-groups.yaml:309]: Plugging in some examples values to understand this:

- 3 out of 10 broker pods in a kafka cluster are not scraping
- 100 * 3 / 10 = 100 * 0.3 = 30   (i.e. 30%)

With a threshold of `3`, the rule would fire when greater than 3% of targets can't be scraped. Is that the intention here?

Comment (ramu11) [reply install/resources/monitoring-cluster/prometheus/prometheus-rules-groups.yaml:309]: Yes, 30% would be fine  with threshold of 3 mins

**Comment by ramu11** [comment]
@david-martin  With a threshold of 3, the rule would fire when greater than 30% of targets can't be scraped

**Review by david-martin — APPROVED** [review]

**Pull request merged**